File: music-application/src/main/java/com/music/sale/application/user/port/outport/UserCommandPort.java
package com.music.sale.application.user.port.outport;

import com.music.sale.domain.user.User;
import com.music.sale.domain.user.UserSocial;
import com.music.sale.domain.user.UserTerms;

public interface UserCommandPort {
    User save(User user);

    UserTerms saveTerms(UserTerms userTerms);
}

File: music-application/src/main/kotlin/com/music/sale/application/user/port/outport/UserCommandPort.kt
package com.music.sale.application.user.port.outport

import com.music.sale.domain.user.User
import com.music.sale.domain.user.UserSocial
import com.music.sale.domain.user.UserTerms

interface UserCommandPort {
    fun save(user: User): User

    fun saveSocialAccount(userSocial: UserSocial): UserSocial

    fun saveTerms(userTerms: UserTerms): UserTerms

    fun updateSocialAccountUserId(
        fromUserId: Long,
        toUserId: Long,
    )

    fun deleteUser(userId: Long)
}
